devito/builtins/arithmetic.py

```diff
@@ -161,3 +161,66 @@ def mmax(f):
         return mr.v.item()
     else:
         raise ValueError("Expected Function, not `%s`" % type(f))
+
+
+def count_nonzero(f):
+    """
+    Retrieve the count of nonzero elements of a (Time)Function.
+
+    Parameters
+    ----------
+    f : array_like or Function
+        Input operand.
+    """
+    ci = dv.ConditionalDimension(name='ci', parent=f.dimensions[-1],
+                                 condition=dv.Ne(f, 0))
+    g = dv.Function(name='g', shape=(1,), dimensions=(ci,), space_order=0, dtype=np.int32)
+
+    # Extra Scalar used to avoid reduction in gcc-5
+    i = dv.types.Scalar(name='i', dtype=np.int32)
+    eqi = dv.Eq(i, 0)
+
+    eq0 = dv.Inc(g[i], 1, implicit_dims=(f.dimensions + (ci,)))
+    op0 = dv.Operator([eqi, eq0], opt=('openmp', {'par-collapse-ncores': 30}))
```

Comment:
Is this really the most efficient way to compute this?

Reply:
wrt par-collpase?

Reply:
No I mean the equation  + conditional but I guess not really simpler way.